### PR TITLE
zlog: retool escaping

### PR DESCRIPTION
**Pull request by hdonnay**
Now with tests!

### CODE REVIEW

**Comment by hdonnay** [comment]
cc @RTann

File: zlog_test.go [hunk 1]
```diff
@@ -12,6 +12,51 @@ import (
...
+		tt := []struct {
+			In   string
+			Want bool
+		}{
+			{"%25", false},
+			{"%2C", false},
+			{"%age", true},
+			{"%", true},
+			{"ꙮ", true},
+			{"%", true},
```
Comment (RTann) [inline zlog_test.go:26]: nit: already tested?

**Review by RTann — COMMENTED** [review]
`$` needs to be escaped?

**Review by RTann — APPROVED** [review]

File: zlog_test.go (hunk 1 above)
Comment (hdonnay) [reply zlog_test.go:26]: I wanted to test specifically that the regexp omits already-escaped values, but catches all the ones we need to escape.

**Comment by hdonnay** [comment]
I'm pretty sure `$` doesn't need special handling. The "value" is specified as
```
baggage-octet          =  %x21 / %x23-2B / %x2D-3A / %x3C-5B / %x5D-7E
                          ; US-ASCII characters excluding CTLs,
                          ; whitespace, DQUOTE, comma, semicolon,
                          ; and backslash
```
and `$` is 0x25, which is in the allowed range.

**Pull request merged**